docs/pages/repo/docs/core-concepts/monorepos/task-dependencies.mdx

```diff
@@ -0,0 +1,189 @@
+and _task_ dependencies, which are different concepts. Workspace dependencies are `dependencies`
+and `devDependencies` in `package.json`, whereas, task dependencies are `dependsOn` key in `turbo.json`.
+</Callout>
+
+The `^` symbol (called a "caret") explicitly declares that the task depends on the task in a
+workspace it depends on.
+
+```jsonc filename="turbo.json"
+{
+  "$schema": "https://turbo.build/schema.json",
+  "pipeline": {
+    "build": {
+      // "A workspace's `build` command depends on its dependencies'
+      // and devDependencies' `build` commands being completed first"
+      "dependsOn": ["^build"],
+    }
+  }
+}
+```
+
+With the configuration above, if an app installs a package from another workspace, the package's
+`build` script will always run before the app's `build` script.
+
+## From arbitrary workspaces
+
+Sometimes, you may want to create a workspace-task depend on another
+workspace-task. This can be especially helpful for repos migrating from `lerna`
+or `rush`, where tasks are run in separate phases by default. Sometimes these
+configurations make assumptions that cannot be expressed in a simple `pipeline`
+configuration, as seen above. Or you may just want to express sequences of tasks
+between applications or microservices when using `turbo` in CI/CD.
+
+For these cases, you can express these relationships in your `pipeline`
+configuration using the `<workspace>#<task>` syntax. The example below describes
+the `deploy` script of a `frontend` application that depends on the `deploy` and
+`health-check` scripts of `backend`, as well as the `test` script of a `ui`
+workspace:
+
+```jsonc filename="turbo.json"
+{
+  "$schema": "https://turbo.build/schema.json",
+  "pipeline": {
+    // Explicit workspace-task to workspace-task dependency
+    "frontend#deploy": {
+      "dependsOn": ["ui#test", "backend#deploy", "backend#health-check"]
+    }
+  }
+}
+```
+
+This explicit configuration for `frontend#deploy` may seem to conflict with the
+`test` and `deploy` task configurations, but it does not. Since `test` and
+`deploy` do not have dependencies on other workspaces (e.g. `^<task>`), they can
+execute any time after their workspace's `build` and `test` scripts have
+finished.
+
+<Callout>
+  Notes:
+
+1. Although this `<workspace>#<task>` syntax is a useful escape hatch, we
+   generally recommend using it for deployment orchestration tasks such as
+   health checks, rather than build-time dependencies, so that Turborepo can
+   optimize these tasks more efficiently
+1. Workspace-tasks do not inherit cache configuration. You must redeclare
+   [`outputs`](/repo/docs/reference/configuration#outputs) at the moment.
+1. `<workspace>` must match the `name` key in the workspace's `package.json` or
+   the task will be ignored.
+
+</Callout>
+
+## Dependencies outside of a task
+
+When your task has topological dependencies that are outside of that given task,
```

Comment:
We should define `topological` here

Reply:
went a little further and explained it even more.